Add support for `ImageGenerationTool` `output_compression` and `output_format` for Vertex AI Gemini image models

h0rv opened the pull request:
As of now, these options are only available via vertex-ai, not google-gla

DouweM commented on tests/models/test_google.py:
```diff
@@ -3694,6 +3694,110 @@ async def test_google_image_generation_auto_size_raises_error(google_provider: G
+
+    # Test output_format rejection
+    agent = Agent(model, builtin_tools=[ImageGenerationTool(output_format='png')], output_type=BinaryImage)
+    with pytest.raises(ValueError, match='output_mime_type parameter is not supported in Gemini API'):
```
Instead of hitting errors, let's ignore the field silently as we would've done before. We've already documented it only works with Vertex, so we can check `self.system == 'google-vertex'` inside `GoogleModel`

DouweM commented on pydantic_ai_slim/pydantic_ai/builtin_tools.py:
```diff
@@ -301,6 +301,7 @@ class ImageGenerationTool(AbstractBuiltinTool):
     Supported by:
 
     * OpenAI Responses. Only supported for 'png' and 'webp' output formats.
+    * Google (Vertex AI only). Only supported for 'jpeg' output format.
```
Please update the docs as well

h0rv commented:
Updated to match Vertex AI documentation for compression quality:
- Changed `ImageGenerationTool.output_compression` default to `None`.
- `GoogleModel` now omits the compression parameter when `None`, allowing Vertex AI to use its documented default of `75`.
- `OpenAIChatModel` preserves its existing behavior by defaulting to `100` when the tool's value is `None`.
- Updated docstrings and tests to reflect these provider-specific defaults.

h0rv commented:
https://platform.openai.com/docs/guides/image-generation#customize-image-output

also fixed openai doc:

<img width="1452" height="406" alt="image" src="https://github.com/user-attachments/assets/1ee6d580-5ab9-469f-9a42-e6fc036b87b3" />

DouweM commented on pydantic_ai_slim/pydantic_ai/models/google.py:
```diff
@@ -358,6 +361,46 @@ async def request_stream(
+                        f'Google image generation `output_compression` must be between 0 and 100. '
+                        f'Got: {output_compression}.'
+                    )
+                if tool.output_format not in (None, 'jpeg'):
```
Does the `None` here mean that if `output_compression` is set but not `output_format`, the used output format automatically becomes JPEG? Then we should update the "default" wording in the docstring above.

DouweM commented on pydantic_ai_slim/pydantic_ai/builtin_tools.py:
```diff
@@ -295,12 +295,13 @@ class ImageGenerationTool(AbstractBuiltinTool):
 
-    * OpenAI Responses. Only supported for 'png' and 'webp' output formats.
+    * OpenAI Responses. Only supported for 'jpeg' and 'webp' output formats. Default: 100.
+    * Google (Vertex AI only). Only supported for 'jpeg' output format. Default: 75.
```
Please update this section: https://ai.pydantic.dev/builtin-tools/#provider-support_4
